rdr_service/services/consent/files.py

```diff
@@ -410,25 +421,56 @@ def _get_confirmation_check_elements(self):
 
 
 class VibrentPrimaryConsentUpdateFile(PrimaryConsentUpdateFile):
-    _SIGNATURE_PAGE = 13
+    FIRST_VERSION_END_DATE = datetime(2020, 11, 1)
+
+    def __init__(self, *args, consent_date: datetime, **kwargs):
+        super(VibrentPrimaryConsentUpdateFile, self).__init__(*args, **kwargs)
+
+        # In Sep 2020, the content of the update consent changed. Versions prior ot that were essentially just
+        # copies of the Primary consent file. If the given file is close enough to the switch-over date then
+        # treat it as a normal consent.
+        self.wrapped_consent_file = None
+        if consent_date < self.FIRST_VERSION_END_DATE and not PrimaryConsentUpdateFile.pdf_has_update_text(self.pdf):
+            self.wrapped_consent_file = VibrentPrimaryConsentFile(*args, **kwargs)
+
+    def _get_signature_page(self):
+        return self.pdf.get_page_number_of_text([
+            ('Do you agree to this updated consent?', '¿Está de acuerdo con este consentimiento actualizado?')
+        ])
 
     def _get_signature_elements(self):
-        return self.pdf.get_elements_intersecting_box(
-            Rect.from_edges(left=150, right=400, bottom=155, top=160),
-            page=self._SIGNATURE_PAGE
-        )
+        if self.wrapped_consent_file:
+            return self.wrapped_consent_file._get_signature_elements()
+        else:
+            return self.pdf.get_elements_intersecting_box(
+                Rect.from_edges(left=150, right=400, bottom=155, top=160),
+                page=self._get_signature_page()
+            )
 
     def _get_date_elements(self):
-        return self.pdf.get_elements_intersecting_box(
-            Rect.from_edges(left=130, right=400, bottom=110, top=115),
-            page=self._SIGNATURE_PAGE
-        )
+        if self.wrapped_consent_file:
+            return self.wrapped_consent_file._get_date_elements()
+        else:
+            return self.pdf.get_elements_intersecting_box(
+                Rect.from_edges(left=130, right=400, bottom=110, top=115),
+                page=self._get_signature_page()
+            )
 
-    def _get_agreement_check_elements(self):
-        return self.pdf.get_elements_intersecting_box(
-            Rect.from_edges(left=38, right=40, bottom=676, top=678),
-            page=self._SIGNATURE_PAGE
-        )
+    def is_agreement_selected(self):
+        if self.wrapped_consent_file:
```

Comment:
Is there a ETA on when we start processing missing consent check marks errors for consents other than GROR?  Will we revalidate all the primary consents for that check?